org.elysium/src/org/elysium/validation/LilyPondIssueCodesProvider.java

```diff
@@ -0,0 +1,30 @@
+package org.elysium.validation;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.validation.IssueSeveritiesProvider;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+
+public class LilyPondIssueCodesProvider extends IssueSeveritiesProvider{
+
+	private LoadingCache<Resource, List<String>> cache = CacheBuilder.newBuilder().expireAfterAccess(3, TimeUnit.SECONDS)
+			.maximumSize(4).build(new CacheLoader<Resource, List<String>>(){
```

Comment:
Sorry, one last comment: are you sure about these cache expiry and size settings? Please document why they give optimal results. Thanks!